apps/settings/js/panels/root/panel.js

```diff
@@ -158,3 +133,29 @@ define(function(require) {
     });
   };
 });
+/**
+ * Update the sim and NFC related items
+   based on mozMobileConnections and Device.
+ */
+
+var updateSimItemsAndNfc = function rph_refrehsSimItems() {
```

Comment:
and still this is wrong approach from architectural POV. This application already have the dedicated object:
 ```'...that helps to enable/disable and to show card state information for telephony-related items from the root in the setting app.'``` (see file: js/telephony_items_handler.js)
To provide required behaviour you just need to slightly update logic in this file.
Regarding last 2 lines which manipulates with 'NFC' item, you shall put them into the panel's ``onInit()`` or ``on BeforeShow()`` handlers.